src/msg/async/EventKqueue.cc

```diff
@@ -199,13 +199,17 @@ int KqueueDriver::resize_events(int newsize)
   ldout(cct,30) << __func__ << " kqfd = " << kqfd << "newsize = " << newsize 
                 << dendl;
   if(newsize > sav_max) {
-    sav_events = (struct SaveEvent*)realloc( sav_events, 
-                    sizeof(struct SaveEvent)*newsize);
-    if (!sav_events) {
+    void *_realloc = NULL;
+    _realloc = realloc(sav_events,
+                          sizeof(struct SaveEvent)*newsize);
+    if (_realloc == NULL) {
```

Comment:
@atheism yes, we need to free() the allocated mem chunk if we are able to recover from the failed call `realloc()` .

> For the caller, we should add error handler to handle the error properly.

i am expecting to the fix in this PR. but i am afraid it would not be a trivial change. hence the suggestion.